src/core/src/core_logic/PatchInstaller.py

```diff
@@ -283,7 +350,137 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
+                else:
+                    packages_in_batch.append(packages[index])
+
+            if len(already_installed_packages) > 0:
+                self.composite_logger.log("Following packages are already installed. Should have got installed as dependent package of some other package " + str(already_installed_packages))
+
+            if len(skip_packages) > 0:
+                self.composite_logger.log("[Skipping packages " + str(skip_packages) + " - requires Ubuntu Advantage for Infrastructure with Extended Security Maintenance]")
+
+            if len(packages_in_batch) == 0:
+                continue
+
+            remaining_time = maintenance_window.get_remaining_time_in_minutes()
+
+            # point in time status
+            progress_status = self.progress_template.format(str(datetime.timedelta(minutes=remaining_time)), str(self.attempted_parent_update_count), str(self.successful_parent_update_count), str(self.failed_parent_update_count), str(installed_update_count - self.successful_parent_update_count),
+                                                            "Processing batch index: " + str(batch_index) + ", Number of packages: " + str(len(packages_in_batch)) + "\nProcessing packages: " + str(packages_in_batch))
+            self.composite_logger.log(progress_status)
+
+            if maintenance_window.is_package_install_time_available(self.reboot_manager, remaining_time, len(packages_in_batch)) is False:
+                self.composite_logger.log("Stopped installing patches in batches as it is past the maintenance window cutoff time for installing in batches." +
+                                           " Batch Index: {0}, remaining time: {1}, number of packages in batch: {2}".format(batch_index, remaining_time, str(len(packages_in_batch))))
+                maintenance_window_batch_cutoff_reached = True
+                break
+
+            packages_and_dependencies = package_manager.include_dependencies(packages_in_batch)
+            self.composite_logger.log("Packages including dependencies in the batch installing are: " + str(packages_and_dependencies))
+
+            packages_dependencies_to_install = []
+            package_dependency_versions = []
+
+            for package in packages_and_dependencies:
+                if package not in all_packages:
+                    continue
+                packages_dependencies_to_install.append(package)
+                version = package_versions[packages.index(package)] if package in packages else Constants.DEFAULT_UNSPECIFIED_VALUE
+                package_dependency_versions.append(version)
+
+            # multilib resolution for yum
+            if package_manager.get_package_manager_setting(Constants.PKG_MGR_SETTING_IDENTITY) == Constants.YUM:
+                for package in packages_in_batch:
+                    self.add_arch_dependencies_yum(package_manager, package, packages, package_versions, packages_dependencies_to_install, package_dependency_versions)
```

Comment:
Generalize into the package manager to keep PatchInstaller generic.

Reply:
Done